### DISCUSSION TITLE
problem inserting new line mini.completion (or it could be the lsp?)

### DISCUSSION THREAD

**Discussion by amirongit** [nvim-mini/mini.nvim#1054] [Q&A] [answered]
so imagine the completion option contains a new line (for example I'm overriding a method from the parent class in python, the completion consists of function signature, new line, and then super().methodname())
but the new line isn't inserted correctly and I have no Idea why! (also can't search google because it seems that google ignores the symbols in the search bar or gives them specific meanings!)
any way!
desired behavior:
1 - I accept the completion option
2 - I get this

```
async def find_peer(self, host: Node, identifier: UniversalPeerIdentifier) -> Peer:
    return await super().find_peer(host, identifier)
```
actual behavior:
1 - I accept the completion option
2 - I get this

```
async def find_peer(self, host: Node, identifier: UniversalPeerIdentifier) -> Peer:�    return await super().find_peer(host, identifier)
```

**Comment by amirongit** [comment]
the new line character in the editor is shown like this: ```^@```

**Comment by echasnovski** [comment, marked as answer]
> so imagine the completion option contains a new line ...

That's the core of the issue right at the start :) Multiline completion candidates are not supported (albeit suggested a long time ago in vim/vim/#2505). And newline symbols in completion candidates indeed are shown as `^@`.

As 'mini.completion' is designed to rely on built-in completion capabilities as much as possible, I am afraid this functionality should be supported there first.

**Comment by amirongit** [comment]
well, can you think of a patch or work around?
@echasnovski

**Reply by echasnovski** [reply]
I am not really comfortable around Vim C code. And as far as I understand, it *has* to be fixed there first.

**Reply by amirongit** [reply]
I meant maybe a workaround that we could do in our configurations in your mind
for example, somehow replacing the "^@" with a new line on every accept event (implemented in cfg rather than code)
anyway, ty for the great plugins ;)

**Reply by echasnovski** [reply]
I *think* I looked into this during development of 'mini.completion'. Also looked at it briefly again and can not see a way to workaround it using *only* built-in completion capabilities (which is what 'mini.completion' is implemented with).
